src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PollingWildCardChangeToken.cs

```diff
@@ -143,7 +143,8 @@ private bool CalculateChanges()
         /// <returns>The <see cref="DateTime"/> that the file was last modified.</returns>
         protected virtual DateTime GetLastWriteUtc(string path)
         {
-            return File.GetLastWriteTimeUtc(Path.Combine(_directoryInfo.FullName, path));
+            string filePath = Path.Combine(_directoryInfo.FullName, path);
+            return FileSystemInfoHelper.ResolveFileLinkTarget(filePath)?.LastWriteTimeUtc ?? File.GetLastWriteTimeUtc(filePath);
```

Comment:
Are we concerned about following the symlink and allocating a FileInfo with every call here?
I think both [windows](https://github.com/ericstj/sample-code/blob/3053b0795ae65030c1f30002704d4d0a8444e435/symLinkConfig/WindowsSymlinkHelper.cs#L53-L72) and [unix](https://github.com/ericstj/sample-code/blob/3053b0795ae65030c1f30002704d4d0a8444e435/symLinkConfig/UnixSymlinkHelper.cs#L22-L29) offer APIs that can let us check this without explicitly following the link.  I know we didn't expose static APIs like that but here might be a place to measure to see if that's important.  Maybe see how much this is actually called in a typical ASP.NET app.

Reply:
As per the perf. of the specific case reported in the issue, which uses `AddJsonFile` (only watches one file) with `reloadOnChange: true`, `PollingFileChangeToken.GetLastWriteTimeUtc` is being called once every 4 seconds and every time the file changes.

Reply:
Looks to me like this is even being called within a method which calls this: https://github.com/dotnet/runtime/blob/08233bac350b19ab2501deafbb3d6683621fe1bd/src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/Internal/MatcherContext.cs#L58

That's doing allocations of similar objects on a much larger scale, so probably not an issue in this particular use case, however it's worth considering that there could be a similar use case where the caller does care about allocation.  There's really no good reason to allocate an *Info object in this case and we have other APIs on File that avoid that.  IIRC those APIs were created for folks who do lots of enumeration and want it lower overhead interaction with file properties.

Reply:
Given the other allocations down this code path I think it's fine to accept this for now.  We can lookout for feedback around this change though.